collections/_posts/2025-01-18-GradientStepDenoiser.md

```diff
@@ -0,0 +1,218 @@
+
+## Variational methods
+
+A very common way of solving inverse problems is to formulate them as optimization problems, more specifically minimizing a cost function, often called *energy*:
+
+$$x^* = \arg \min_x f(x, y) + g(x)$$
+
+where $$f$$ is the data-fidelity term, and $$g$$ is the regularization.
+You can think of $$f$$ as the part that measures how well the data $$x$$ explains the observation $$y$$, while $$g$$ encodes the prior knowledge.
+
+Let's take a very simple example. We know that our data is corrupted by Gaussian noise, and that our solution is supposed to be sparse. In this case, the correct data-fildelity term is $$f(x, y) = \|x - y\|_2^2$$, and a regularization that promotes sparsity would be $$g(x) = \|x\|_1 $$.
+
+Now we can use all the literature on convex optimization to solve our problem.
+However, in many cases our regularization $$g$$ is not differentiable everywhere, meaning that we can't use a regular Gradient Descent algorithm.
+This is for instance the case for the $$L_1$$ regularization we just chose, but it is also the case for the *Total Variation* regularization, which is still very common in imaging inverse problems.
+Indeed this regularization promotes *piecewise constant* solutions, which describes well-enough natural images with sharp edges.
+
+## Proximal operator
+
+How do we get around the non-differentiability of our objective function ?
+Well some very smart mathematicians have a developed a special tool in order to minimize non-smooth function.
+This tool is called the *proximal map* or *proximal operator* and you can think of it as a sort of smooth projection.
+
+![The moreau enveloppe of the L1 norm](/collections/images/GradientStep/moreau.jpg)
+
+Since applied mathematicians love optimization, this tool is itself formulated as an optimization problem:
+
+$$\text{prox}_{\tau f} (x) = \arg \min_u f(u) + \frac{1}{2 \tau} \| x - u\|^2$$
+
+As you can see when computing the prox of a function, the goal is to minimize the function while staying close to a given point $x$.
+Long story short, there are closed form solutions for the proximal operators of a lot of usual functions.
+And more precisely, most usual non-smooth regularization function that one might use in inverse problems admit closed-form proximal operators.
+This is for instance the case of our previous example with the $$L_1$$ norm, but also the famous *Total Variation* regularization and others.
+
+A typical algorithm to solve the previous minimization problem would then be the *Forward-Backward* splitting, or *Proximal Gradient Descent*.
+
+$$x^{n+1} = \text{prox}_{\tau g} \circ \left( \text{Id} - \tau \nabla f \right) (x^{n}) $$
+
+As you can see it's just a regular gradient descent on the smooth data-fidelity term $f$ and a proximal step on the non-smooth regularization $g$.
+There are a lot of other algorithms using the proximal operator, but this is the most basic one.
+
+---
+
+# Plug-and-Play
```

Comment:
In this part and the following maybe state clearly which paper and contribution you are talking about.